src/model/GameObject.js

```diff
@@ -225,6 +226,13 @@ GameObject.prototype.scheduleTimer = function scheduleTimer(options, type, key)
 			rc.run(
 				function timerCall() {
 					log.trace({options: options}, '%s call', type);
+					if (self.stale) {
+						if (options.interval) {
+							clearInterval(self.gsTimers[type][key].handle);
+						}
+						delete self.gsTimers[type][key];
+						throw new Error('stale object');
```

Comment:
No, there is no other cleanup code as of now; this problem just became apparent recently with NPC movement.
It's not "expected" in the sense that it's unavoidable, but it can happen in case of bugs, and this should make sure it does not cause major havoc (particularly with intervals).
The request context error handling actually catches the thrown Error (and logs it as an ERROR), so apart from the log message level I think this is doing pretty much what you suggest.